src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyStringComparison.cs

```diff
@@ -82,6 +84,18 @@ public override void Initialize(AnalysisContext analysisContext)
                         return;
                     }
 
+                    if (iqueryableType != null)
+                    {
+                        var ancestorInvocation = invocationExpression.GetAncestor<IInvocationOperation>(OperationKind.Invocation);
+
+                        // If we are in a IQueryable context, it's possible that the underlying call doesn't have the comparison
+                        // option so we want to bail-out to be safe.
+                        if (ancestorInvocation != null && iqueryableType.Equals(ancestorInvocation.Type?.OriginalDefinition))
```

Comment:
My first thought was to not stop on the first invocation ancestor but it doesn't seem to be needed.